src/firefly/js/visualize/ui/ExtractionDialog.jsx

```diff
@@ -395,6 +386,118 @@ function LineExtractionPanel({canCreateExtractionTable, pv, pvCnt}) {
+    let pt0= undefined;
+    let pt1= undefined;
+    const {dataHeight,dataWidth}= plot ?? {};
+    if (!plot) return {pt0,pt1};
+
+    const validLine= !isNaN(parseInt(line)) && !isNaN(Number(line));
+    const validCol= !isNaN(parseInt(col)) && !isNaN(Number(col));
+
+    if ( (isLine && validLine) || (!isLine && validCol) ) {
+        if (isLine && line>dataHeight) return {pt0,pt1};
+        if (!isLine && col>dataWidth) return {pt0,pt1};
+        pt0= isLine ? makeImagePt(0,line) : makeImagePt(col, 0);
+        pt1= isLine ? makeImagePt(dataWidth-1,line) : makeImagePt(col, dataHeight-1);
+    }
+    return {pt0,pt1};
+}
+
+
+function ExtractWholeLine({plot}) {
+    const dirVal= useFieldGroupValue('direction')[0]() ?? 'X';
+    const col= useFieldGroupValue('column')[0]();
+    const line= useFieldGroupValue('line')[0]();
+    const isLine= dirVal==='X';
+    const {dataHeight,dataWidth}= plot ?? {};
+    const singleLine= dataWidth===1 || dataHeight===1;
+
+    const {pt0, pt1}= !singleLine ? getPts(isLine,line,col,plot) :
+        dataWidth===1 ? getPts(false, '',0,plot) : getPts(true,0,'',plot);
+        
+
+    useEffect(() => {
+        if (!plot) return;
+        dispatchModifyCustomField( ExtractLineTool.TYPE_ID,
+            {
+                newFirst: pt0,
+                newCurrent: pt1,
+                plotId: plot.plotId,
+            },
+            plot.plotId);
+
+    }, [line,col,dirVal]);
+
+
+    if (!plot) return;
+
+    const direction= (
+        <ListBoxInputField
+                    fieldKey='direction'
+                    options={[ {label: 'Line', value: 'X'}, {label: 'Column', value: 'Y'} ]}
+                    slotProps={{
+                        input: { variant:'plain', sx:{minHeight:'unset'} }
+                    }}
+                    initialState= {{ value:dirVal}}
+                    tooltip='Select line or column' label=''  />
+    );
+
+
+    const extractButton= (
+        <Button
+                variant= 'solid'
+                color= 'primary'
+                disabled={Boolean(!pt0 && !pt1)}
+                onClick={()=> {
+                    const dl= getDrawLayerByType(getDlAry(), ExtractLineTool.TYPE_ID);
+                    if (!dl) return;
+                    addLineDistAttributesToPlots(dl,plot.plotId,{pt0,pt1});
+                    // dispatchAttributeChange({plotId:plot.plotId, toAllPlotsInPlotView:true, overlayColorScope:false,
+                    //     changes: { [PlotAttribute.ACTIVE_DISTANCE]: sel, [PlotAttribute.EXTRACTION_DATA]: true, }
+                    // });
+                }} >
+            Extract
+        </Button>
+    );
+
+    const text= !singleLine  ?
+        `Extract a whole line or column from ${plot.dataWidth}x${plot.dataHeight} image` :
+        dataWidth===1 ?
+            `Extract a whole column from ${plot.dataWidth}x${plot.dataHeight} image` :
+            `Extract a whole line ${plot.dataWidth}x${plot.dataHeight} image`;
```

Comment:
Based on my comment above, the second statement here could also be enclosed in parentheses if you want. But since it's already multi-line, it's quite readable.